src/govuk/components/character-count/template.njk

```diff
@@ -35,9 +39,14 @@
     attributes: params.attributes
   }) }}
   {%- set fallbackHintLength = params.maxwords or params.maxlength %}
-  {%- set fallbackHintDefault = 'You can enter up to %{count} ' + ('words' if params.maxwords else 'characters') %}
+  {#
+    Use the fact that `html` has precedence over `text` to override content
+    when no limit is set and reserve vertical space to avoid layout shifting
+    by injecting a non-breakable space
+  #}
   {{ govukHint({
-    text: (params.fallbackHintText or fallbackHintDefault) | replace('%{count}', fallbackHintLength),
+    text: ((fallbackHintText) | replace('%{count}', fallbackHintLength) if not hasNoLimit),
+    html: ('&nbsp;' if hasNoLimit),
```

Comment:
This does feel a little messy. Given this is (hopefully) something of an edge case, I'm wondering if we just live with the layout shift as the hint gets injected.

Alternatively, what if we added a class to make the hint `visibility: hidden` if we don't know the count, then remove it once the JS fires?